src/core/main.c

```diff
@@ -2312,6 +2325,12 @@ int main(int argc, char *argv[]) {
         if (r < 0)
                 goto finish;
 
+        if (!skip_setup && arg_early_core_pattern) {
```

Comment:
please move this further down, so that this is only executed when `arg_action == ACTION_RUN`. After all we don't want this to run if people invoke us with `--version` or suchlike. (there's already an if block for that further down, just add it to the beginning of that)

Also, this needs to be conditionalized more tightly: this should only be done in system mode (i.e. when getpid_cached() == 1) and not in a container...

I think it would make sense to make this a function of its own, given that these conditionalizations are quite a few

Reply:
If we make sure that `getpid_cached() == 1`, there's no need to also check for `arg_action == ACTION_RUN`, right ?

Reply:
I'm asking because it seems a good idea to redirect coredump as early as possible.